src/libraries/Hooks.sol

```diff
@@ -67,35 +73,79 @@ library Hooks {
             );
     }
 
-    function shouldCallBeforeInitialize(IHooks self) internal pure returns (bool) {
-        return uint256(uint160(address(self))) & BEFORE_INITIALIZE_FLAG != 0;
+    function validateHooksResponse(bytes4 selector, bytes4 expectedSelector) internal pure {
+        if (selector != expectedSelector) {
+            revert InvalidHookResponse();
+        }
```

Comment:
to add: noop handling here

Reply:
this refactor will be strange with NoOp because we have to run the clear currentHook logic before returning